contracts/RelayedDocumentStore.sol

```diff
@@ -0,0 +1,129 @@
+  *          The resulting signature should be in hex format following the Ethereum signature encoding.
+  *          For example:
+  *           - const message = "0x3a267813bea8120f55a7b9ca814c34dd89f237502544d7c75dfd709a659f6330";
+  *             const nonce = RelayedDocumentStore.signerNonce(trustedSigner);
+  *             const address = RelayedDocumentStore.address; // 0x8CdaF0CD259887258Bc13a92C0a6dA92698644C0
+  *             const messageHash = web3.utils.soliditySha3(message, nonce, address); // 0x2bd7dce779d8d21e3266e847f21ce8160a53e6a78a65a5045f72b18081f7fc69
+  *             const signature = await web3.eth.sign(messageHash, signer); // 0xf5731da089e7d9532aad446f9f92cabf3ce39c5cde8ba541cb26ed17b69ab8870a411a66ad8316d7b7478e1c1be05ca840ecfc8cc7468beb9c2832be4dc8fb1e1b
+ */
+contract RelayedDocumentStore is Initializable, Ownable {
+  using ECDSA for bytes32;
+
+  address private trustedSigner; // Account that is signing *messages*
+  DocumentStore public documentStore;
+
+  mapping(address => uint256) public signerNonce;
+
+  event DocumentIssued(bytes32 indexed document);
+  event DocumentRevoked(bytes32 indexed document);
+  event RelayedMessage(address indexed relayer, address indexed trustedSigner, bytes4 functionIdentifier);
+
+  function initialize(
+    string memory _name,
+    address _trustedSigner,
+    address _relayer
+  )
+    public
+    initializer
+  {
+    // Owner of this contract is the relayer
+    require(_relayer != address(0), "RelayedDocumentStore: relayer is the zero address");
+    super.initialize(_relayer);
+
+    require(_trustedSigner != address(0), "RelayedDocumentStore: trustedSigner is the zero address");
+    trustedSigner = _trustedSigner;
+
+    // Owner of the DocumentStore is this contract enforcing trustedSigner sign-off
+    documentStore = new DocumentStore();
+    documentStore.initialize(_name, address(this));
+  }
+
+  function bulkIssueRelayed(
+    bytes32[] memory documents,
+    bytes32 documentsHash,
+    bytes memory signature
+  )
+    public
+    onlyOwner
+    onlyTrustedSigner(documentsHash, signature)
+  {
+    documentStore.bulkIssue(documents);
+  }
+
+  function bulkRevokeRelayed(
+    bytes32[] memory documents,
+    bytes32 documentsHash,
+    bytes memory signature
+  )
+    public
+    onlyOwner
+    onlyTrustedSigner(documentsHash, signature)
+  {
+    documentStore.bulkRevoke(documents);
+  }
+
+  function issueRelayed(
+    bytes32 document,
+    bytes memory signature
+  )
+    public
+    onlyOwner
+    onlyTrustedSigner(document, signature)
+  {
+    documentStore.issue(document);
+  }
+
+  function revokeRelayed(
+    bytes32 document,
+    bytes memory signature
+  )
+    public
+    onlyOwner
+    onlyTrustedSigner(document, signature)
+  {
+    documentStore.revoke(document);
+  }
+
+  /**
+   * @dev Throws if the message is not signed by the trusted signer or nonce incorrect.
+   */
+  modifier onlyTrustedSigner(bytes32 message, bytes memory signature) {
```

Comment:
(!!) there's nothing tying the hash issued with the signature that's passed in separately, so the relayer actually can issue whatever he wants given a valid signature from the trustedSigner

Reply:
The `message` is actually the hash in the modifier here so I believe that should tie it together? I am going to add `msg.sig` though or perhaps go with full `calldata` to be signed though.